regression-test/suites/query_p0/sql_functions/test_predicate.groovy

```diff
@@ -0,0 +1,48 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+suite("test_predicate") {
+    sql """drop table if exists t1;"""
+    sql """
+            create table t1 (
+                k1 int null,
+                k2 int null,
+                k3 int null,
+                k4 int null
+            )
+            duplicate key (k1)
+            distributed BY hash(k1) buckets 3
+            properties("replication_num" = "1");
+        """
+    sql """ insert into t1 values(1,2,3,null); """
+
+    qt_null_or_null "select * from t1 where abs(k1)=null or abs(k1)=null;"
+    qt_true_or_null "select * from t1 where abs(k1)=1 or abs(k1)=null;"
```

Comment:
the condition in where predicate is always not null, it is not same as the decription you provide in the issue.

Reply:
> the condition in where predicate is always not null, it is not same as the decription you provide in the issue.

predicate's nullable_mode is depend on arguments